minions/async/src/main/java/io/github/microcks/minion/async/producer/MQTTProducerManager.java

```diff
@@ -67,27 +74,29 @@ public void create() throws Exception {
 
    /**
     * Create a IMqttClient and connect it to the server.
+    * 
     * @return A new IMqttClient implementation initialized with configuration properties.
     * @throws Exception in case of connection failure
     */
    protected IMqttClient createClient() throws Exception {
       MqttConnectOptions options = new MqttConnectOptions();
-      if (mqttUsername != null && mqttUsername.length() > 0
-            && mqttPassword != null && mqttPassword.length() > 0) {
+      if (mqttUsername != null && mqttUsername.length() > 0 && mqttPassword != null && mqttPassword.length() > 0) {
          options.setUserName(mqttUsername);
          options.setPassword(mqttPassword.toCharArray());
       }
       options.setAutomaticReconnect(true);
       // Set clean session to false as we're reusing the same clientId.
       options.setCleanSession(false);
-      options.setConnectionTimeout(10);
-      IMqttClient publisher = new MqttClient("tcp://" + mqttServer, "microcks-async-minion");
+      options.setConnectionTimeout(20);
+      options.setKeepAliveInterval(10);
```

Comment:
As I'm quite inexperienced with MQTT, can you explain the importance of this 2 settings?

Reply:
It turns out they were not the cause of my problem, I changed these values in an attempt to resolve my connection problems. 
Anyway - I think these are suitable defaults.

setKeepAliveInterval - MQTT client sends ping control packets and receive a response in regular intervals in case no other comms have taken place (such as a message published or received) to keep the connection alive
Paho default is quite high, a lot of other clients seem to opt for lower interval, e.g. MQTT Box uses 10s.
setConnectionTimeout - how long the client waits after to establish a server connection.
Paho default is 20 seconds, I think 20-30 seconds is adequate.

Reply:
Thanks for explanation.